pcdsdevices/targets.py

```diff
@@ -260,3 +267,795 @@ def move(self, nxspaces, nyspaces, wait=False):
+        if not hasattr(self.x.presets, 'add_hutch'):
+            raise AttributeError('No folder setup for motor presets. '
+                                 'Please add a location to save the positions '
+                                 'to, using setup_preset_paths from '
+                                 'pcdsdevices.interface to save the position.')
+
+        print('\nSetting coordinates for (0, 0) top left corner: \n')
+        self.tweak()
+        pos = [self.x.position, self.y.position]
+        print('\nSetting coordinates for (0, M) top right corner: \n')
+        self.tweak()
+        pos.extend([self.x.position, self.y.position])
+        print('\nSetting coordinates for (N, M) bottom right corner: \n')
+        self.tweak()
+        pos.extend([self.x.position, self.y.position])
+        print('\nSetting coordinates for (N, 0) bottom left corner: \n')
+        self.tweak()
+        pos.extend([self.x.position, self.y.position])
+        # create presets
+        # corner (0, 0)
+        self.x.presets.add_hutch(value=pos[0], name="x_top_left")
+        self.y.presets.add_hutch(value=pos[1], name="y_top_left")
+        # corner (0, M)
+        self.x.presets.add_hutch(value=pos[2], name="x_top_right")
+        self.y.presets.add_hutch(value=pos[3], name="y_top_right")
+        # corner (M, N)
+        self.x.presets.add_hutch(value=pos[4], name="x_bottom_right")
+        self.y.presets.add_hutch(value=pos[5], name="y_bottom_right")
+        # corner (N, 0)
+        self.x.presets.add_hutch(value=pos[6], name="x_bottom_left")
+        self.y.presets.add_hutch(value=pos[7], name="y_bottom_left")
+
+    def get_presets(self):
+        """
+        Get the saved presets if any.
+
+        Examples
+        --------
+        >>> xy.get_presets()
+        ((0, 0),
+        (9.99999999999998, 0),
+        (9.99999999999998, -9.99999999999998),
+        (-6.38378239159465e-16, -9.99999999999998))
+
+        Returns
+        -------
+        coord : tuple
+            Four coordinate positions.
+            (top_left, top_right, bottom_right, bottom_left)
+        """
+        try:
+            top_left = (self.x.presets.positions.x_top_left.pos,
+                        self.y.presets.positions.y_top_left.pos)
+            # corner (0, M)
+            top_right = (self.x.presets.positions.x_top_right.pos,
+                         self.y.presets.positions.y_top_right.pos)
+            # corner (M, N)
+            bottom_right = (self.x.presets.positions.x_bottom_right.pos,
+                            self.y.presets.positions.y_bottom_right.pos)
+            # corner (N, 0)
+            bottom_left = (self.x.presets.positions.x_bottom_left.pos,
+                           self.y.presets.positions.y_bottom_left.pos)
+            return top_left, top_right, bottom_right, bottom_left
+        except Exception:
+            logger.warning('Could not get presets, try to set_presets.')
+
+
+class XYGridStage(XYTargetGrid):
```

Comment:
Which features of `XYTargetGrid` are you using in `XYGridStage`? To me it looks like you're defining new `move_to` functions, for example, and not using the parent's resources. The grid is computed differently too, the parent using this comp/skew parameter, for example, that is being ignored here.

Am I missing a reason that this needs to be a subclass?

Reply:
I think I was initially using the `x_spacing` and `y_spacing` - and it made more sense to inherit from this class, but as of now I am not using those, I am only using the `StageStack` portion of it. So I actually don't need to inherit from this class. I think I should move the twek method with the set and get presets to my object instead.

The only thing that I am using now is the `StageStack` from this class, which allows the x_motor and y-motor to either be created from a PV or a motor object, I can inherit from `StageStack` only if I need to keep this functionality

Reply:
So you are correct, I am not really using the parent's resources - thank you for pointing this out

Reply:
Okay, that makes sense- stuff evolves over time as we code/test!

Reply:
Took the inheritance out of this object (393ee51)